### ISSUE TITLE
No Perception Expander Lines on Kobo Libra 2

### ISSUE THREAD

**Issue by slobberingant** [koreader/koreader#10691]
* KOReader version: 2023.06.1
* Device: Kobo Libra 2

#### Issue
Perception expander lines in the speed reader module not showing while reading books.
Have tried the following:
- Changed line thickness to 1, 5, 10, 99 etc.
- Changed margin incase screen size was incorrectly configured. 1, 3, 5, 20, 100, 400
- Line color intensity changed. 1, 5, 10.
- Tried manually changing settings in the config file. These changes were shown on the ereader.
- Enabling / disabling.
- Different books.
- Restarting etc.
- Perception expander lines work on koreader 2023.06.1 on my linux system.

Verbose (debug mode) crash.log file attached.
Please advise how I could find more information if required.
[crash.log](https://github.com/koreader/koreader/files/12038122/crash.log)

**Comment by NiLuJe** [comment]
Yeah, I have no idea how this thing is even supposed to work, but that's unsurprising, it doesn't ever actually request a rferesh.

(This thing is *very* weird and clunky).

**Comment by hius07** [comment]
It works good on Kindle right now.

**Comment by Frenzie** [comment]
Afaik it works on Kobo too (at least my old H2O) but it's been quite a while since I checked.

**Comment by NiLuJe** [comment]
> it doesn't ever actually request a refresh.

Yeah, I suppose the intent is to bake the draw into the page *before* the refresh actually goes through.

Anyway, I'll double-check the timings on my end, but this thing is weird.

**Comment by Frenzie** [comment]
Still works for me.

**Comment by NiLuJe** [comment]
> Anyway, I'll double-check the timings on my end, but this thing is weird.

It uses the `view_modules` thingy, so, yeah, the ordering is completely sound.

TL;DR: I can't see any way for this to go wrong except with nonsensical settings ;).

**Comment by NiLuJe** [comment]
And, yeah, @worksforme ;).

Unless the Libra 2 is weirdly allergic to thin gray lines, no clue.

**Comment by slobberingant** [comment]
It's all good. I didn't realise that the margin value was a value between 0 - 1. I could have sworn this value was in pixels before.
Sometimes one just needs to be told that they are the problem.
Appreciate the responses.
